### ISSUE TITLE
Build error - ubuntu

### ISSUE THREAD

**Issue by D4nte** [status-im/nim-waku#866]
Trying to build nim-waku master on ubuntu machine.

```
$ sudo apt install g++ make
$ make
Building: Nim compiler
Building: install/usr/lib/libbacktrace.a
Building: install/usr/lib/libbacktracenim.a
Building: build/wakunode1
[NimScript] exec: nim c --out:build/wakunode1 -d:chronicles_log_level=DEBUG -d:chronosStrictException --verbosity:0 --hints:off -d:release waku/v1/node/wakunode1.nim
Building: build/sim1
[NimScript] exec: nim c --out:build/quicksim -d:chronicles_log_level=INFO -d:chronosStrictException --verbosity:0 --hints:off -d:release waku/v1/node/quicksim.nim
[NimScript] exec: nim c --out:build/start_network -d:chronicles_log_level=DEBUG -d:chronosStrictException --verbosity:0 --hints:off -d:release waku/v1/node/start_network.nim
Building: build/example1
[NimScript] exec: nim c --out:build/example -d:chronicles_log_level=DEBUG -d:chronosStrictException --verbosity:0 --hints:off -d:release examples/v1/example.nim
Building: build/wakunode2
[NimScript] exec: nim c --out:build/wakunode2 -d:chronicles_log_level=DEBUG --verbosity:0 --hints:off -d:release waku/v2/node/wakunode2.nim
/home/fryorcraken/src/nim-waku/waku/v2/protocol/waku_relay.nim(60, 8) Warning: use {.base.} for base methods; baseless methods are deprecated [UseBase]
/home/fryorcraken/src/nim-waku/waku/v2/protocol/waku_relay.nim(75, 8) Warning: use {.base.} for base methods; baseless methods are deprecated [UseBase]
/home/fryorcraken/src/nim-waku/waku/v2/protocol/waku_relay.nim(81, 8) Warning: use {.base.} for base methods; baseless methods are deprecated [UseBase]
/home/fryorcraken/src/nim-waku/waku/v2/protocol/waku_store/waku_store.nim(10, 6) Warning: imported and not used: 'algorithm' [UnusedImport]
/home/fryorcraken/src/nim-waku/waku/v2/utils/wakuswitch.nim(6, 7) Warning: imported and not used: 'byteutils' [UnusedImport]
/home/fryorcraken/src/nim-waku/waku/v2/utils/wakuswitch.nim(4, 6) Warning: imported and not used: 'sequtils' [UnusedImport]
/home/fryorcraken/src/nim-waku/waku/v2/utils/wakuswitch.nim(10, 23) Warning: imported and not used: 'dnsresolver' [UnusedImport]
/home/fryorcraken/src/nim-waku/waku/v2/node/discv5/waku_discv5.nim(9, 3) Warning: imported and not used: 'config' [UnusedImport]
Killed
stack trace: (most recent call last)
/home/fryorcraken/src/nim-waku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(416, 18)
/home/fryorcraken/src/nim-waku/waku.nimble(68, 15) wakunode2Task
/home/fryorcraken/src/nim-waku/waku.nimble(35, 8) buildBinary
/home/fryorcraken/src/nim-waku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(260, 7) exec
/home/fryorcraken/src/nim-waku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(260, 7) Error: unhandled exception: FAILED: nim c --out:build/wakunode2 -d:chronicles_log_level=DEBUG --verbosity:0 --hints:off -d:release waku/v2/node/wakunode2.nim [OSError]
make: *** [Makefile:139: wakunode2] Error 1
```

**Comment by jm-clius** [comment]
I cannot reproduce locally. I use Ubuntu and all make targets pass on my machine. Perhaps caused by some submodule inconsistencies? You could try to run `make clean update` manually to get (hopefully) to a supported repo state.

**Comment by D4nte** [comment]
I did `make update` and did not help.

Trying `make clean update`. I am using `Ubuntu 20.04.4 LTS`.

**Comment by D4nte** [comment]
Did not help.

Do note the fact that I have to use `OVERRIDE=1` which is also cumbersome

```
fryorcraken@waku1:~/src/nim-waku$ make clean update
 m vendor/libbacktrace-upstream
^---vendor/nim-libbacktrace
 D allocfail.sh
^---vendor/libbacktrace-upstream

You have uncommitted local changes which might be overwritten by the update. Aborting.
If you know better, you can re-run the command with OVERRIDE=1.

make: *** [vendor/nimbus-build-system/makefiles/targets.mk:88: update-test] Error 1
fryorcraken@waku1:~/src/nim-waku$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
fryorcraken@waku1:~/src/nim-waku$ OVERRIDE=1 make clean update
 m vendor/libbacktrace-upstream
^---vendor/nim-libbacktrace
 D allocfail.sh
^---vendor/libbacktrace-upstream
Building: Nim compiler
fryorcraken@waku1:~/src/nim-waku$ make wakunode2
Building: install/usr/lib/libbacktrace.a
Building: install/usr/lib/libbacktracenim.a
Building: build/wakunode2
[NimScript] exec: nim c --out:build/wakunode2 -d:chronicles_log_level=DEBUG --verbosity:0 --hints:off -d:release waku/v2/node/wakunode2.nim
/home/fryorcraken/src/nim-waku/waku/v2/protocol/waku_relay.nim(60, 8) Warning: use {.base.} for base methods; baseless methods are deprecated [UseBase]
/home/fryorcraken/src/nim-waku/waku/v2/protocol/waku_relay.nim(75, 8) Warning: use {.base.} for base methods; baseless methods are deprecated [UseBase]
/home/fryorcraken/src/nim-waku/waku/v2/protocol/waku_relay.nim(81, 8) Warning: use {.base.} for base methods; baseless methods are deprecated [UseBase]
/home/fryorcraken/src/nim-waku/waku/v2/protocol/waku_store/waku_store.nim(10, 6) Warning: imported and not used: 'algorithm' [UnusedImport]
/home/fryorcraken/src/nim-waku/waku/v2/utils/wakuswitch.nim(6, 7) Warning: imported and not used: 'byteutils' [UnusedImport]
/home/fryorcraken/src/nim-waku/waku/v2/utils/wakuswitch.nim(4, 6) Warning: imported and not used: 'sequtils' [UnusedImport]
/home/fryorcraken/src/nim-waku/waku/v2/utils/wakuswitch.nim(10, 23) Warning: imported and not used: 'dnsresolver' [UnusedImport]
Killed
stack trace: (most recent call last)
/home/fryorcraken/src/nim-waku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(416, 18)
/home/fryorcraken/src/nim-waku/waku.nimble(68, 15) wakunode2Task
/home/fryorcraken/src/nim-waku/waku.nimble(35, 8) buildBinary
/home/fryorcraken/src/nim-waku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(260, 7) exec
/home/fryorcraken/src/nim-waku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(260, 7) Error: unhandled exception: FAILED: nim c --out:build/wakunode2 -d:chronicles_log_level=DEBUG --verbosity:0 --hints:off -d:release waku/v2/node/wakunode2.nim [OSError]
make: *** [Makefile:139: wakunode2] Error 1
```

**Comment by D4nte** [comment]
Versions:

```
fryorcraken@waku1:~/src/nim-waku$ make -v
GNU Make 4.2.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
fryorcraken@waku1:~/src/nim-waku$ g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-17ubuntu1~20.04' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-HskZEa/gcc-9-9.3.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04) 
```

**Comment by jm-clius** [comment]
🤔 Ensured that I have clean repo state and again failed to reproduce.

Versions seem aligned with yours as well:

```
GNU Make 4.2.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

```
g++ -v
Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-17ubuntu1~20.04' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-HskZEa/gcc-9-9.3.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04)
```

Should also note that CI on `ubuntu` is passing.
If you have to use `OVERRIDE =1`, perhaps there are some local inconsistencies in your repo? Not sure if you've tried `rm -rf vendor/*` yet?

**Comment by D4nte** [comment]
deleting `vendor/` did not help:

```
fryorcraken@waku1:~/src/nim-waku$ rm -rf vendor/
fryorcraken@waku1:~/src/nim-waku$ make wakunode2
Git submodules not found. Running 'git submodule update --init --recursive'.

Submodule path 'vendor/dnsclient.nim': checked out 'fbb76f8af8a33ab818184a7d4406d9fee20993be'
Submodule path 'vendor/news': checked out 'fadc54f4683869fe9c97fe59e122daa2607abc8e'
Submodule path 'vendor/nim-bearssl': checked out 'ba80e2a0d7ae8aab666cee013e38ff8d33a3e5e7'
Submodule path 'vendor/nim-bearssl/bearssl/csources': checked out 'acc70b1be60a6f321e2da618cd35d901b1a598a4'
Submodule path 'vendor/nim-chronicles': checked out '1682096306ddba8185dcfac360a8c3f952d721e4'
Submodule path 'vendor/nim-chronos': checked out '87197230779002a2bfa8642f0e2ae07e2349e304'
Submodule path 'vendor/nim-confutils': checked out '0435e67832b6bb8dfdf0ddb102903e9d820206d2'
Submodule path 'vendor/nim-dnsdisc': checked out '6aa5603ab405fa0297550c76b26ea0d2bb8736c5'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-bearssl': checked out 'ba80e2a0d7ae8aab666cee013e38ff8d33a3e5e7'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-bearssl/bearssl/csources': checked out 'acc70b1be60a6f321e2da618cd35d901b1a598a4'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-chronicles': checked out '1682096306ddba8185dcfac360a8c3f952d721e4'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-chronos': checked out '519ca463dfd060dfec64a221b8e39e06a70860d3'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-confutils': checked out '0435e67832b6bb8dfdf0ddb102903e9d820206d2'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-eth': checked out '4e2b340af659d959b76160d4e6b39a3c395e9a9a'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-faststreams': checked out '03975dadf6fedb8beb5d1b4d4c5bec8635fd7577'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-http-utils': checked out 'ad6f82e4b7d504ea9b8222c9c01aa1d24ebd4dc8'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-json-rpc': checked out '5a281760803907f4989cacf109b516381dfbbe11'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-json-serialization': checked out '4b8f487d2dfdd941df7408ceaa70b174cce02180'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-secp256k1': checked out 'e092373a5cbe1fa25abfc62e0f2a5f138dc3fb13'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-secp256k1/secp256k1_wrapper/secp256k1': checked out '1766dc808621f8a6e91282a321ddabce4352e624'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-serialization': checked out '37bc0db558d85711967acb16e9bb822b06911d46'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-stew': checked out 'b464505b4dcbe2ef52d19b2cfca8f2be4ebf2c7e'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-testutils': checked out '1f66448152a03e45ebb556c00300e13565cd8adf'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-unittest2': checked out '24eb5e829ed07e71e3a5d09786d5f80aa988874f'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-websock': checked out 'f354dfebe9f45ba33e6ad3b6f3163d516c9d0727'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-zlib': checked out '6bbc67d09e624d69052d62b2353878f491186942'
Submodule path 'vendor/nim-dnsdisc/vendor/nim-zlib/zlib/csources': checked out 'cacf7f1d4e3d44d871b605da3b647f07d718623f'
Submodule path 'vendor/nim-dnsdisc/vendor/nimbus-build-system': checked out 'a1da1f403d9034f0f09582c832863c902751e9fb'
Submodule path 'vendor/nim-dnsdisc/vendor/nimbus-build-system/vendor/Nim': checked out '121628357ec7fae91335bd392f03b0e6dd249319'
Submodule path 'vendor/nim-dnsdisc/vendor/nimbus-build-system/vendor/Nim-csources-v1': checked out 'a8a5241f9475099c823cfe1a5e0ca4022ac201ff'
Submodule path 'vendor/nim-dnsdisc/vendor/nimbus-build-system/vendor/nimble': checked out 'd13f3b8ce288b4dc8c34c219a4e050aaeaf43fc9'
Submodule path 'vendor/nim-dnsdisc/vendor/nimcrypto': checked out 'a5742a9a214ac33f91615f3862c7b099aec43b00'
Submodule path 'vendor/nim-eth': checked out '4e2b340af659d959b76160d4e6b39a3c395e9a9a'
Submodule path 'vendor/nim-faststreams': checked out '03975dadf6fedb8beb5d1b4d4c5bec8635fd7577'
Submodule path 'vendor/nim-http-utils': checked out '40048e8b3e69284bdb5d4daa0a16ad93402c55db'
Submodule path 'vendor/nim-json-rpc': checked out '5a281760803907f4989cacf109b516381dfbbe11'
Submodule path 'vendor/nim-json-serialization': checked out '4b8f487d2dfdd941df7408ceaa70b174cce02180'
Submodule path 'vendor/nim-libbacktrace': checked out '284b3aac05a9d96c27044c389a5d27a84d8e8f4b'
Submodule path 'vendor/nim-libbacktrace/vendor/libbacktrace-upstream': checked out '4f57c999716847e45505b3df170150876b545088'
Submodule path 'vendor/nim-libbacktrace/vendor/whereami': checked out 'f3a86fdf17b49c434a16bb4d9e45a135d4cc25f9'
Submodule path 'vendor/nim-libp2p': checked out '07da14a7a7de467b9d5cfe46eeaba18c8de7a7e0'
Submodule path 'vendor/nim-metrics': checked out '71e0f0e354e1f4c59e3dc92153989c8b723c3440'
Submodule path 'vendor/nim-nat-traversal': checked out '8994b67b07813955c61bebddf4bd2325439c3535'
Submodule path 'vendor/nim-nat-traversal/vendor/libnatpmp-upstream': checked out '4536032ae32268a45c073a4d5e91bbab4534773a'
Submodule path 'vendor/nim-nat-traversal/vendor/miniupnp': checked out 'f5f693876db9b1d1328811ffc4f3ef4903b965ca'
Submodule path 'vendor/nim-secp256k1': checked out 'e092373a5cbe1fa25abfc62e0f2a5f138dc3fb13'
Submodule path 'vendor/nim-secp256k1/secp256k1_wrapper/secp256k1': checked out '1766dc808621f8a6e91282a321ddabce4352e624'
Submodule path 'vendor/nim-serialization': checked out '37bc0db558d85711967acb16e9bb822b06911d46'
Submodule path 'vendor/nim-sqlite3-abi': checked out '07039dd887c4e5b57367a16f4be3c18763be1d7b'
Submodule path 'vendor/nim-stew': checked out 'b464505b4dcbe2ef52d19b2cfca8f2be4ebf2c7e'
Submodule path 'vendor/nim-stint': checked out 'ddfa6c608a6c2a843d7b405f377a22703947267a'
Submodule path 'vendor/nim-testutils': checked out '1f66448152a03e45ebb556c00300e13565cd8adf'
Submodule path 'vendor/nim-unittest2': checked out '24eb5e829ed07e71e3a5d09786d5f80aa988874f'
Submodule path 'vendor/nim-web3': checked out '0f82cf22f37fecd6dedf838c0c592c33d0748ab4'
Submodule path 'vendor/nim-websock': checked out '73edde4417f7b45003113b7a34212c3ccd95b9fd'
Submodule path 'vendor/nim-zlib': checked out '74cdeb54b21bededb5a515d36f608bc1850555a2'
Submodule path 'vendor/nim-zlib/zlib/csources': checked out 'cacf7f1d4e3d44d871b605da3b647f07d718623f'
Submodule path 'vendor/nimbus-build-system': checked out 'bc4791418a3d748c6114e7219d95739ad154ecaa'
Submodule path 'vendor/nimbus-build-system/vendor/Nim': checked out 'c6a9f27b3e36bae9aacec1bd6c37893fb98fd33f'
Submodule path 'vendor/nimbus-build-system/vendor/Nim-csources-v1': checked out 'a8a5241f9475099c823cfe1a5e0ca4022ac201ff'
Submodule path 'vendor/nimbus-build-system/vendor/nimble': checked out 'd13f3b8ce288b4dc8c34c219a4e050aaeaf43fc9'
Submodule path 'vendor/nimcrypto': checked out 'a5742a9a214ac33f91615f3862c7b099aec43b00'
Submodule path 'vendor/rln': checked out '7ac74183f8b69b399e3bc96c1ae8ab61c026dc43'

Building: Nim compiler
make[1]: *** [vendor/nimbus-build-system/makefiles/targets.mk:77: build-nim] Error 1
make: *** [vendor/nimbus-build-system/makefiles/targets.mk:116: vendor/nimbus-build-system/vendor/Nim/bin/nim] Error 2
```

Doing a fresh checkout.

**Comment by D4nte** [comment]
Still failing with a fresh checkout. I am guessing it's because it's a droplet and hence it reaches some capacity limit when compiling.

**Comment by kaiserd** [comment]
@D4nte yes. The smallest droplet is not enough and compiling nim-waku on it will lead to this error. You need at least 2GB of ram or activate swap in the smallest droplet (1GB).
You can also upgrade the smallest droplet to 2Gb, compile, and then downgrade to 1GB.
Shortly upgrading to a very strong one to save time won't work, because they tend to have a different CPU architecture.

**Comment by D4nte** [comment]
Thanks @kaiserd, it makes sense. I'll look into using the later docker image instead.

**Comment by D4nte** [comment]
May I suggest that the doc gets updated to mention the RAM requirements to build as the error is not obvious.

**Comment by jm-clius** [comment]
@kaiserd to add a line or two to the README, after which this issue can probably be closed.

**Comment by kaiserd** [comment]
Done. See PR #889

**Comment by D4nte** [comment]
fixed with #889
